ui/src/scenes/UserProfile/UserProfile.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Container, Row, Col } from 'reactstrap';
+import LocalGraph from '@/components/LocalGraph';
+
+const UserProfile = ({ location }) => {
+  const {
+    email, name, nodeId, __typename,
```

Comment:
I'd say use a [url parameter](https://reacttraining.com/react-router/web/example/url-params) ("/profile/:personId") instead of location.state. That way every profile gets its own URL. See the NeedsContainer component for an example of how to get the :personId url parameter from react-router so you can use it in the component.

Reply:
Yeah, that's what I first wanted to do but at the time of writing there is no way to find a user in the database by their id. The only way to query for a user is by their email and we decided that it wouldn't look so great if the users email was part of the URL. 

And I don't know Cypher well enough to implement querying by ID 😅

Reply:
Done and pushed. The person field in the GraphQL schema now accepts nodeId as an argument in this branch.